src/app/rpmostree-compose-builtin-tree.cxx

```diff
@@ -1079,10 +1095,13 @@ impl_commit_tree (RpmOstreeTreeComposeContext *self,
                            g_variant_ref_sink (v));
     }
 
-  /* Convert metadata hash to GVariant */
+  /* Convert metadata hash tables to GVariants */
   g_autoptr(GVariant) metadata = rpmostree_composeutil_finalize_metadata (self->metadata, self->rootfs_dfd, error);
   if (!metadata)
     return FALSE;
+  g_autoptr(GVariant) detached_metadata = rpmostree_composeutil_finalize_detached_metadata(self->detached_metadata);
+  if (!detached_metadata)
+    return FALSE;
```

Comment:
The error-handling here is incorrect. But I think it probably makes sense to have `rpmostree_composeutil_finalize_detached_metadata` canonicalize to `NULL` if the hashtable is empty so that we don't create detached commit metadata for nothing.

So then here you can just remove the last two lines of this hunk.

Reply:
Fixed, thanks!